tslearn/metrics.py

```diff
@@ -2031,6 +2039,241 @@ def cdist_soft_dtw_normalized(dataset1, dataset2=None, gamma=1.):
     return dists
 
 
+@njit(nogil=True)
+def njit_lcss_accumulated_matrix(s1, s2, eps, delta):
+    """Compute the longest common subsequence similarity score between
+    two time series.
+
+    Parameters
+    ----------
+    s1 : array, shape = (sz1,)
+        First time series.
+
+    s2 : array, shape = (sz2,)
+        Second time series.
+
+    eps : float
+        Matching threshold.
+
+    delta : int
+        Maximum distance in the x-axis (time) for the matching to occur.
+
+    Returns
+    -------
+    lcss_score : float
+        Longest Common Subsequence similarity score between both time series.
+
+    """
+    l1 = s1.shape[0]
+    l2 = s2.shape[0]
+    acc_cost_mat = numpy.full((l1 + 1, l2 + 1), 0)
+
+    for i in range(1, l1 + 1):
+        for j in range(1, l2 + 1):
+            if _local_eucl_dist(s1[i - 1], s2[j - 1]) <= eps\
```

Comment:
we should use the same default inner distance as for Dynamic Time Warping (and maybe allow one to change that upon request)